src/Features/header/components/Sidebar/SidebarSlice.ts

```diff
@@ -0,0 +1,50 @@
+import { createApi } from '@reduxjs/toolkit/query/react';
+import type { BaseQueryFn } from '@reduxjs/toolkit/query/react';
+import axios from 'axios';
+import type { AxiosRequestConfig, AxiosError } from 'axios';
+
+const axiosBaseQuery =
+  (
+    { baseUrl }: { baseUrl: string } = { baseUrl: '' },
+  ): BaseQueryFn<
+    {
+      url: string;
+      method: AxiosRequestConfig['method'];
+      data?: AxiosRequestConfig['data'];
+      params?: AxiosRequestConfig['params'];
+    },
+    unknown,
+    unknown
+  > =>
+  async ({ url, method, data, params }) => {
+    try {
+      const result = await axios({ url: baseUrl + url, method, data, params });
+      return { data: result.data.data };
+    } catch (axiosError) {
+      const err = axiosError as AxiosError;
+      return {
+        error: {
+          status: err.response?.status,
+          data: err.response?.data || err.message,
+        },
+      };
+    }
+  };
+
+const api = 'http://localhost:3000/';
+export const sidebarSlice = createApi({
```

Comment:
Hm, I see naming weirdness. We are talking about cities, but then you name your slice `sidebarSlice`. Sidevar is not descriptive. If don't know what sidebar has, you won't guess what it does. For example, it can be `bookingSlice` or `reservationSlice` because sidebar will contain much more information besides cities